test/integration/image-optimizer/test/index.test.js

```diff
@@ -1108,4 +1108,46 @@ describe('Image Optimizer', () => {
 
     setupTests(true)
   })
+
+  describe('content disposition header', () => {
+    beforeAll(async () => {
+      const json = JSON.stringify({
+        images: {
+          domains: ['image-optimization-test.vercel.app'],
+        },
+      })
+      nextConfig.replace('{ /* replaceme */ }', json)
+      appPort = await findPort()
+      app = await launchApp(appDir, appPort)
+    })
+
+    afterAll(async () => {
+      await killApp(app)
+      nextConfig.restore()
+      await fs.remove(imagesDir)
+    })
+
+    it.each`
+      url                                                                 | contentDisposition
+      ${'/test.jpg'}                                                      | ${`inline; filename="test.webp"`}
+      ${'/test.png'}                                                      | ${`inline; filename="test.webp"`}
+      ${'/test.svg'}                                                      | ${`inline; filename="test.svg"`}
+      ${'https://image-optimization-test.vercel.app/api/frog.jpg'}        | ${`inline; filename="frog.webp"`}
+      ${'https://image-optimization-test.vercel.app/api/frog.png'}        | ${`inline; filename="frog.webp"`}
+      ${'https://image-optimization-test.vercel.app/api/vercel.svg'}      | ${`inline; filename="vercel.svg"`}
+      ${'https://image-optimization-test.vercel.app/png-as-octet-stream'} | ${`inline; filename="png-as-octet-stream.webp"`}
```

Comment:
I'm not so sure about changing the octet stream to webp 🤔

Reply:
the `content-type` that is returned in the response is `image/webp`

Reply:
Oh yes, that was added in #26705

I think keeping `.webp` extension is fine here. Lets add it to the existing test here:

https://github.com/vercel/next.js/blob/2a5dd0b19b0665cfcabde12ad13e9a39096967fd/test/integration/image-optimizer/test/index.test.js#L355-L356